cassandra/cluster.py

```diff
@@ -1235,7 +1235,7 @@ def shutdown(self):
 
         self.control_connection.shutdown()
 
-        for session in self.sessions:
+        for session in tuple(self.sessions):
```

Comment:
I don't know that I'm convinced -- it's not a big cost to do this, and asynchronous operations could create a session during shutdown. Yes, that's a bug, but I think that the "can't edit sessions while iterating over it" error may not be the best way to communicate that. Am I on the right track?